shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -353,10 +276,14 @@ - (void)loadAOTData:(NSString*)assetsDir {
 
 - (void)setViewController:(FlutterViewController*)controller {
   _viewController = controller;
-  _mainOpenGLContext = controller.flutterView.openGLContext;
+
+  if (controller) {
+    [_openGLRenderer attachToFlutterView:controller.flutterView];
+  }
+
   if (!controller && !_allowHeadlessExecution) {
+    [_openGLRenderer clearResourceContext];
     [self shutDownEngine];
-    _resourceContext = nil;
   }
```

Comment:
There is a codepath here (`nil` `controller`, `_allowHeadlessExecution` true) wher you aren't updating the renderer.

Reply:
Fixed it to clear the resource context when we don't have a view controller.